tools/MultiProjectValidator/ProjectAnalyzer.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Generic;
+using Microsoft.DotNet.ProjectModel;
+using MultiProjectValidator.AnalysisRules;
+
+namespace MultiProjectValidator
+{
+    public class ProjectAnalyzer
+    {
+
+        public static ProjectAnalyzer Create(List<ProjectContext> projectContexts)
+        {
+            // Any Additional rules would be added here
+            var rules = new List<IAnalysisRule>
+            {
+                new DependencyMismatchRule()
+            };
+
+            return new ProjectAnalyzer(rules, projectContexts);
+        }
+
+        private List<ProjectContext> projectContexts;
+        private List<IAnalysisRule> rules;
+        
+        private ProjectAnalyzer(List<IAnalysisRule> rules, List<ProjectContext> projectContexts)
+        {
+            this.rules = rules;
+            this.projectContexts = projectContexts;
+        }
```

Comment:
@brthor Checkout these [coding guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) for private fields.

>We use `_camelCase` for internal and private fields and use readonly where possible. Prefix instance fields with _, static fields with s_ and thread static fields with t_.